rust/agama-lib/src/storage/client/dasd.rs

```diff
@@ -66,6 +76,149 @@ impl<'a> DASDClient<'a> {
         Ok(introspect.contains("org.opensuse.Agama.Storage1.DASD.Manager"))
     }
 
+    pub async fn get_config(&self) -> Result<DASDConfig, ServiceError> {
+        // TODO: implement
+        Ok(DASDConfig::default())
+    }
+
+    pub async fn set_config(&self, config: DASDConfig) -> Result<(), ServiceError> {
+        // at first probe to ensure we work on real system info
+        self.probe().await?;
+        self.config_activate(&config).await?;
+        self.config_format(&config).await?;
+        self.config_set_diag(&config).await?;
+
+        Ok(())
+    }
+
+    async fn config_activate(&self, config: &DASDConfig) -> Result<(), ServiceError> {
+        let pairs = self.config_pairs(config).await?;
+        let to_activate: Vec<&str> = pairs
+            .iter()
+            .filter(|(system, _config)| system.enabled == false)
+            .filter(|(_system, config)| {
+                config.state.clone().unwrap_or_default() == DASDDeviceState::Active
+            })
+            .map(|(system, _config)| system.id.as_str())
+            .collect();
+        self.enable(&to_activate).await?;
+
+        let pairs = self.config_pairs(config).await?;
```

Comment:
Any reason to read the devices again? I wonder whether it was possible to use a single call to `config_pairs`.

Reply:
Ah, I planned to have config_pair produce iterator instead of vector, so in that case it will be consumed. But maybe it will be more efficient to just use original vector.